File: _legacy/controller/nests/delete_item.php
<?php
namespace Core;
use Controller\Main, Controller\files\UtilForFiles;
use Exception;

if (!defined('__API_GOOSE__')) exit();

/**
 * delete nest
 *
 * @var Goose|Connect $this
 */

try
{
  // check and set srl
  if (($srl = (int)($this->params['srl'] ?? 0)) <= 0)
  {
    throw new Exception(Message::make('error.notFound', 'srl'));
  }

  // connect db
  $this->model->connect();

  // check access
  $token = Main::checkAccessItem($this, (object)[
    'table' => 'nests',
    'srl' => $srl,
  ]);

  // remove articles
  $articles = $this->model->getItems((object)[
    'table' => 'articles',
    'field' => 'srl',
    'where' => 'nest_srl='.$srl,
  ]);
  if (count($articles->data ?? []) > 0)
  {
    foreach($articles->data as $k=>$v)
    {
      // remove thumbnail image
      UtilForFiles::removeThumbnailImage($this, $v->srl);
      // remove files
      UtilForFiles::removeAttachFiles($this, $v->srl, 'articles');
    }
    // remove articles
    $this->model->delete((object)[
      'table' => 'articles',
      'where' => 'nest_srl='.$srl,
    ]);
  }

  // remove categories
  $categoriesCount = $this->model->getCount((object)[
    'table' => 'categories',
    'where' => 'module="article" and target_srl='.$srl,
  ])->data;
  if ($categoriesCount > 0)
  {
    $this->model->delete((object)[
      'table' => 'categories',
      'where' => 'target_srl='.$srl,
    ]);
  }

  // remove nest
  $output = Main::delete($this, (object)[
    'table' => 'nests',
    'srl' => $srl,
  ]);

  // set output
  if ($token) $output->_token = $token->jwt;

  // disconnect db
  $this->model->disconnect();

  // output data
  return Output::result($output);
}
catch (Exception $e)
{
  if (isset($this->model)) $this->model->disconnect();
  return Error::result($e->getMessage(), $e->getCode());
}

File: src/endpoints/app/delete_item.py
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token
from . import __types__ as types
from ..nest import __libs__ as nest_libs
from ..article import __libs__ as article_libs

async def delete_item(params: dict = {}, req = None, _db: DB = None, _token = None):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # set params
        params = types.DeleteItem(**params)

        # checking token
        token = checking_token(req, db) if not _token else _token

        # set where
        where = [ f'srl = {params.srl}' ]

        # check item
        count = db.get_count(
            table_name = Table.APP.value,
            where = where,
        )
        if count <= 0: raise Exception('Item not found.', 204)

        # delete articles
        articles = db.get_items(
            table_name=Table.ARTICLE.value,
            fields=[ 'srl' ],
            where=[ f'app_srl = {params.srl}' ],
        )
        if articles and len(articles) > 0:
            for article in articles: article_libs.delete(db, article.get('srl'))

        # delete nests
        nests = db.get_items(
            table_name=Table.NEST.value,
            fields=[ 'srl' ],
            where=[ f'app_srl = {params.srl}' ],
        )
        if nests and len(nests) > 0:
            for nest in nests: nest_libs.delete(db, nest.get('srl'))

        # delete item
        db.delete_item(
            table_name=Table.APP.value,
            where=where,
        )

        # set result
        result = output.success({
            'message': 'Complete delete app.',
        }, _req=req)
    except Exception as e:
        result = output.exc(e, _req=req)
    finally:
        if not _db and db: db.disconnect()
        return result
